### ISSUE TITLE
"Reply placeholder" is not focusable and is missing a `role="button"`

### ISSUE THREAD

**Issue by tarjeiba** [flarum/framework#3659]
### Current Behavior

The `ReplyPlaceholder` component is inaccessible via keyboard navigation.

### Steps to Reproduce

1. Open a discussion
2. Try to tab navigate to the reply placeholder
3. alternatively, try to use other method to navigate to reply via keyboard

### Expected Behavior

Since the `article` element is clickable (i.e. interactable), it should also be accessible via keyboard.

### Environment

N/A, I found this as a user, not admin.

### Possible Solution

I haven't written any jsx before, but it looks like `tabindex="0"` and `role="button"` should be added to the article tag in `ReplyPlaceholder.js`

```javascript
      <article className="Post ReplyPlaceholder" onclick={reply}>
        <header className="Post-header">
          {avatar(app.session.user, { className: 'PostUser-avatar' })} {app.translator.trans('core.forum.post_stream.reply_placeholder')}
        </header>
      </article>
```

According to [mdn: ARIA role button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) we then also need to add listeners for `space` and `enter` keydown-listeners
